boardswarm-cli/src/main.rs

```diff
@@ -242,6 +242,76 @@ enum AuthCommand {
     Modify(AuthModifyArg),
 }
 
+#[derive(Clone, Debug)]
+enum ItemArg {
+    Id(u64),
+    Name(String),
+}
+
+async fn item_lookup(
+    arg: ItemArg,
+    item_type: ItemType,
+    mut client: Boardswarm,
+) -> Result<u64, anyhow::Error> {
+    let item_kind = match item_type {
+        ItemType::Device => ("Device", "device"),
+        ItemType::Console => ("Console", "console"),
+        ItemType::Actuator => ("Actuator", "actuator"),
+        ItemType::Volume => ("Volume", "volume"),
+    };
+    match arg {
+        ItemArg::Id(id) => Ok(id),
+        ItemArg::Name(name) => {
+            let mut items = client.list(item_type).await?;
+
+            let name_components: Vec<_> = name.split('@').collect();
+            if name_components.len() > 2 {
+                return Err(anyhow!("Too many @ in {} name", item_kind.1));
+            }
+
+            /*
+             * first try finding a single exact match for the "name"
+             * part of the user-supplied "name[@instance]"
+             */
+            items.retain(|i| i.name == name_components[0]);
+            match items.len() {
+                0 => Err(anyhow!("{} not found", item_kind.0)),
+                1 => Ok(items[0].id),
+                /* More matching entries found, try disambiguating by matching instance name */
```

Comment:
with the earlier suggestion you wouldn't need any of of this unless you want to prefer the local one on multiple matches, though i'm not sure if that's a wanted semantic

Reply:
I am afraid we _have to_ prefer local one on multiple matches because  there's no "@local". So if there are ttyS0, ttyS1 and ttyS0@remote, then the user saying "ttyS0" must mean the local one. And it's quite logical, because if the user wanted the remote one, they would say "ttyS0@remote".